src/error_chain.rs

```diff
@@ -378,6 +380,31 @@ macro_rules! impl_error_chain_processed {
             }
         }
 
+       #[cfg(has_error_description_deprecated)]
+       impl_error_chain_kind! {
+            /// The kind of an error.
+            #[derive(Debug)]
+            pub enum $error_kind_name {
+                $(
+                    $(#[$meta_links])*
+                    $link_variant(e: $link_kind_path) {
+                        description(e.description())
+                        display("{}", e)
+                    }
+                ) *
+
+                $(
+                    $(#[$meta_foreign_links])*
+                    $foreign_link_variant(err: $foreign_link_error_path) {
+                        description("")
+                        display("{}", err)
+                    }
+                ) *
+
+                $($error_chunks)*
+            }
+        }
```

Comment:
This code duplication is unfortunate. How about you instead add this helper macro:

```rust
#[cfg(has_error_description_deprecated)]
#[macro_export(local_inner_macros)]
macro_rules! call_to_deprecated_description {
    ($e:ident) => { "" };
}

#[cfg(not(has_error_description_deprecated))]
#[macro_export(local_inner_macros)]
macro_rules! call_to_deprecated_description {
    ($e:ident) => { ::std::error::Error::description($e) };
}
```

And change this one line to:

```rust
description(call_to_deprecated_description!(err))
```

I quickly tested locally and it should work. Not sure if others would consider this overkill just to avoid 20 lines of duplicate code...

Reply:
I like this idea.